internal/api/impl/auth/oidc.go

```diff
@@ -93,24 +134,32 @@ func (e *oIDCEndpoint) buildAuthHandler() echo.HandlerFunc {
 }
 
 func (e *oIDCEndpoint) buildCodeExchangeHandler() echo.HandlerFunc {
-	// TODO(cegarcia): Make a user synchronization operation on the database
-	marshalUserinfo := func(w http.ResponseWriter, r *http.Request, tokens *oidc.Tokens[*oidc.IDTokenClaims], state string, rp rp.RelyingParty, info *oidc.UserInfo) {
+	marshalUserinfo := func(w http.ResponseWriter, r *http.Request, tokens *oidc.Tokens[*oidc.IDTokenClaims], state string, rp rp.RelyingParty, info *oidcUserInfo) {
 		redirectURI := r.URL.Query().Get("redirect_uri")
 		if redirectURI == "" {
 			redirectURI = "/"
 		}
-		login := info.Subject
+		// We don´t forget to set the issuer before making any sync in the database.
+		info.issuer = e.issuer
+
+		user, err := e.svc.SyncUser(info)
+		if err != nil {
+			w.WriteHeader(500)
```

Comment:
what does this value (`500`) mean ?

Reply:
I confess that putting 500 is a magic number yeah haha but common everybody knows what 500 means. I will put something more clear.

Btw I should maybe put a 401 instead. I will stick with the existing from native auth

Reply:
ah it is 400 bad request that is returned in the native. Should I put the same ? Do you return vague 400 intentionally for security reason ?

Reply:
Aligned /login endpoints. 400 if bad creds and 500 otherwise. Note that for external providers, bad creds is handled in their UI with red message in the html